### PR TITLE
Configurable Repositories, Split state into it's own containers, Fixed OS detection

### PR DESCRIPTION
**Pull request by johnkeates**
A few things are fixed now:
- Proper map.jinja and defaults.yaml interaction
- Configurable version for the repository so you can install newer versions
- Configurable configuration directory in case you want to use a custom logstash but still manage it
- OS detection, select based on Debian as a family so Debian and Ubuntu are both fixed regarding the adm group (so logstash can actually access the logs)

### CODE REVIEW

File: logstash/repo.sls [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
-logstash-repo-key:
-  cmd.run:
-    - name: wget -O - http://packages.elasticsearch.org/GPG-KEY-elasticsearch | apt-key add -
-    - unless: apt-key list | grep 'Elasticsearch (Elasticsearch Signing Key)'
+beats_repo_https_apt_support:
+  pkg.installed:
+    - name: apt-transport-https
 
-logstash-repo:
+logstash_repo:
```
Comment (babilen) [inline logstash/repo.sls]: You use dashes in other state IDs, why this change here? In fact the state ID of the one being used for RedHat is `logstash-repo`

Comment (johnkeates) [inline logstash/repo.sls]: Not sure, must have made a mistake, let's fix that.

**Comment by johnkeates** [comment]
It's fixed.

**Comment by kense2660** [comment]
I've tested logstash-formula with your changes to install and configure latest Logstash 2.3.1 but discovered a little mistake. When using `elasticsearch` plugin in your `output` configuration the **host** setting needs to be defined as **hosts** and as an array because it's obselete from version >2.0. Logstash won't work if not used correctly. Is it possible for you to make these changes?

See https://www.elastic.co/guide/en/logstash/2.3/plugins-outputs-elasticsearch.html

Example:
```
elasticsearch {
     hosts => ["1.2.3.4"]
}
```

**Comment by johnkeates** [comment]
Wasn't that the case in the older version (of this formula) as well? Anyhow, I agree, this needs a fix.
